src/pages/[platform]/build-a-backend/data/custom-business-logic/search-and-aggregate-queries/index.mdx

```diff
@@ -0,0 +1,497 @@
+    inlinePolicies: {
+      openSearchPipelinePolicy: new iam.PolicyDocument({
+        statements: [
+          new iam.PolicyStatement({
+            actions: ["es:DescribeDomain"],
+            resources: [
+              openSearchDomain.domainArn,
+              openSearchDomain.domainArn + "/*",
+            ],
+            effect: iam.Effect.ALLOW,
+          }),
+          new iam.PolicyStatement({
+            actions: ["es:ESHttp*"],
+            resources: [
+              openSearchDomain.domainArn,
+              openSearchDomain.domainArn + "/*",
+            ],
+            effect: iam.Effect.ALLOW,
+          }),
+          new iam.PolicyStatement({
+            effect: iam.Effect.ALLOW,
+            actions: [
+              "s3:GetObject",
+              "s3:AbortMultipartUpload",
+              "s3:PutObject",
+              "s3:PutObjectAcl",
+            ],
+            resources: [s3BucketArn, s3BucketArn + "/*"],
+          }),
+          new iam.PolicyStatement({
+            effect: iam.Effect.ALLOW,
+            actions: [
+              "dynamodb:DescribeTable",
+              "dynamodb:DescribeContinuousBackups",
+              "dynamodb:ExportTableToPointInTime",
+              "dynamodb:DescribeExport",
+              "dynamodb:DescribeStream",
+              "dynamodb:GetRecords",
+              "dynamodb:GetShardIterator",
+            ],
+            resources: [tableArn, tableArn + "/*"],
+          }),
+        ],
+      }),
+    },
+    managedPolicies: [
+      iam.ManagedPolicy.fromAwsManagedPolicyName(
+        "AmazonOpenSearchIngestionFullAccess"
+      ),
+    ],
+  }
+);
+
+```
+For the S3 bucket, follow standard security practices: block public access, encrypt data at rest, and enable versioning.
+
+The IAM role should allow the OpenSearch Ingestion Service (OSIS) pipelines to assume it. Grant specific OpenSearch Service permissions and also provide DynamoDB and S3 access. You may customize permissions to follow the principle of least privilege.
+
+### Step 3b: OpenSearch Service Pipeline
+
+Define the pipeline construct and its configuration.
+
+When using OpenSearch, you can define the index template or mapping in advance based on your data structure, which allows you to set data types for each field in the document. This approach can be incredibly powerful for precise data ingestion and search. For more information on index mapping/templates, please refer to [OpenSearch documentation](https://opensearch.org/docs/latest/im-plugin/index-templates/).
+
+Customize the `template_content` JSON-representation to define the data structure for the ingestion pipeline.
+
+``` ts title="amplify/backend.ts"
+
+// Define OpenSearch index mappings
+const indexName = "todo";
+
+const indexMapping = {
+  settings: {
+    number_of_shards: 1,
+    number_of_replicas: 0,
+  },
+  mappings: {
+    properties: {
+      id: {
+        type: "keyword",
+      },
+      isDone: {
```

Comment:
Nit: in the model mentioned above, this is called `done`, I'm not sure if that would be a problem.

Reply:
`done` is a boolean, hence it's referenced as `isDone`. I can see the expected results.